plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/HiveMetastoreBackedDeltaLakeMetastore.java

```diff
@@ -203,11 +206,16 @@ public ProtocolEntry getProtocol(ConnectorSession session, TableSnapshot tableSn
     }
 
     @Override
-    public String getTableLocation(SchemaTableName table)
+    public String getTableLocation(SchemaTableName tableName)
+    {
+        Table table = getTable(tableName.getSchemaName(), tableName.getTableName())
+                .orElseThrow(() -> new TableNotFoundException(tableName));
+        return getTableLocation(table);
+    }
+
+    public static String getTableLocation(Table table)
     {
-        Map<String, String> serdeParameters = getTable(table.getSchemaName(), table.getTableName())
-                .orElseThrow(() -> new TableNotFoundException(table))
-                .getStorage().getSerdeParameters();
+        Map<String, String> serdeParameters = table.getStorage().getSerdeParameters();
```

Comment:
Getting the table location to be used for invalidating the cache seems a bit different from what is being used when populating the transaction log cache.

See 
https://github.com/trinodb/trino/blob/fe5a925d5a58bd184be97874fa41aa44f72f59fd/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/HiveMetastoreBackedDeltaLakeMetastore.java#L152

Reply:
I knew. I followed the same logic invalidating the cache, not there.